src/topology/separation.lean

```diff
@@ -720,11 +720,11 @@ eq_of_nhds_eq_nhds (by simp [nhds_induced, ← set.image_singleton, hf.preimage_
 is the discrete topology on `X`. -/
 lemma discrete_topology_induced {X Y : Type*} [tY : topological_space Y] [discrete_topology Y]
   {f : X → Y} (hf : function.injective f) : @discrete_topology X (topological_space.induced f tY) :=
-begin
-  constructor,
-  rw discrete_topology.eq_bot Y,
-  exact induced_bot hf
-end
+by apply discrete_topology.mk; by rw [discrete_topology.eq_bot Y, induced_bot hf]
```

Comment:
Does `⟨by rw [discrete_topology.eq_bot Y, induced_bot hf]⟩` work here ?

Reply:
Doesn't seem to work:
```
failed to synthesize type class instance for
X : Type u_1,
Y : Type u_2,
tY : topological_space Y,
_inst_1 : discrete_topology Y,
f : X → Y,
hf : function.injective f
⊢ topological_space X
```